nuxeo-core/nuxeo-core/src/main/resources/OSGI-INF/retention-and-hold-contrib.xml

```diff
@@ -10,4 +10,32 @@
       order="1" />
   </extension>
 
+  <extension target="org.nuxeo.ecm.platform.scheduler.core.service.SchedulerRegistryService" point="schedule">
+    <schedule id="findRetentionExpired">
+      <eventId>findRetentionExpired</eventId>
+      <!-- every hour -->
+      <cronExpression>0 0 * * * ?</cronExpression>
+    </schedule>
+  </extension>
+
+  <extension target="org.nuxeo.ecm.core.event.EventServiceComponent" point="listener">
+    <listener name="findRetentionExpired" async="true"
+      class="org.nuxeo.ecm.core.security.RetentionExpiredFinderListener">
+      <event>findRetentionExpired</event>
+    </listener>
+  </extension>
+
+  <extension target="org.nuxeo.ecm.core.bulk" point="actions">
+    <!-- NOT httpEnabled -->
+    <action name="retentionExpired" bucketSize="100" batchSize="25" />
+  </extension>
+
+  <extension target="org.nuxeo.runtime.stream.service" point="streamProcessor">
+    <streamProcessor name="retentionExpired"
+      class="org.nuxeo.ecm.core.security.RetentionExpiredAction"
+      logConfig="bulk" defaultConcurrency="1" defaultPartitions="1">
+      <policy name="default" maxRetries="3" delay="500ms" maxDelay="10s" continueOnFailure="false" />
```

Comment:
Here we could apply a default retry policy of 15min, afaiu this action is called periodically with the same NXQL query the execution is not critical and in case of failure there should be no inconsistancy, so we could continue on failure:
 <policy name="status" maxRetries="20" delay="1s" maxDelay="60s"        continueOnFailure="true" />